scrapbook/models.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+"""
+notebook.py
+
+Provides the Notebook wrapper objects for scrapbook
+"""
+from __future__ import unicode_literals
+import os
+import operator
+import itertools
+import collections
+import pandas as pd
+
+from six import string_types
+from IPython.display import display as ip_display, Markdown
+# We lean on papermill's readers to connect to remote stores
+from papermill.iorw import load_notebook_node
+
+from .translators import registry as translator_registry
+from .exceptions import ScrapbookException
+
+
+GLUE_OUTPUT_PREFIX = 'application/scrapbook.scrap+'
+RECORD_OUTPUT_PREFIX = 'application/papermill.record+'
+DATA_OUTPUT_PREFIXES = [
+    GLUE_OUTPUT_PREFIX,
+    # Backwards compatibility
+    RECORD_OUTPUT_PREFIX
+]
+
+
+def merge_dicts(dicts):
+    iterdicts = iter(dicts)
+    outcome = next(iterdicts).copy()
+    boom = False
+    for d in iterdicts:
+        outcome.update(d)
+    return outcome
+
+
+class Notebook(object):
+    """
+    Representation of a notebook.
+
+    Parameters
+    ----------
+    node : `nbformat.NotebookNode`, str
+        a notebook object, or a path to a notebook object
+    """
+
+    def __init__(self, node_or_path, translators=None):
```

Comment:
Using nb instead of implementation `node_or_path` the method simplifies to:

```python
def __init__(self, nb, translators=None):
        if isinstance(nb, string_types):
            if not nb.endswith(".ipynb"):
                raise ValueError("Requires an '.ipynb' file extension. Provided path: '{}'".format(nb))
            self.path = nb
            self.node = load_notebook_node(nb)
        else:
            self.path = ''
            self.node = nb
        self.translators = translators or translator_registry
         # Memoized traits
        self._scraps = None
        self._frames = None
```